spec/text-document-spec.coffee

```diff
@@ -2,28 +2,51 @@ Point = require "../src/point"
 TextDocument = require "../src/text-document"
 
 describe "TextDocument", ->
-  document = null
-
-  beforeEach ->
-    document = new TextDocument
+  [document, layer] = []
 
   describe "::buildDisplayLayer()", ->
     describe "::slice(start, end)", ->
       it "returns the content between the start and end points", ->
+        document = new TextDocument(softWrapColumn: 10, tabLength: 4)
```

Comment:
I think the best approach is to just expose classes to atom/atom that we'll use to replace existing components of the editor. So I think we should export `TextDisplayDocument`, `TextDocument` etc. This library is for internal consumption for the time being anyway, so we might as well expose an API that makes it as easy as possible to integrate into Atom.